app/controllers/courses_controller.rb

```diff
@@ -1,17 +1,22 @@
 class CoursesController < ApplicationController
   include SessionsHelper
-  before_action :logged_in_user, only: %i[index new create]
-  before_action :admin_user, only: %i[new create]
+  before_action :logged_in_user,
+                only: %i[index show new create edit update destroy]
+  before_action :set_course, only: %i[show edit update destroy]
+  before_action :admin_user, only: %i[new create edit update destroy]
 
-  # GET /users
-  def index; end
+  # GET /courses
+  def index
+    @courses = Course.where.not(deleted: true).paginate(page: params[:page],
```

Comment:
em dùng soft delete à ?
move cái `not(deleted: true)` vào scope . có thể dùng default_scope những chỗ khác không cần gọi cái này nữa

Reply:
Với trường hợp dùng default_scope thì nhớ search thêm về `unscoped`